src/assignment1.js

```diff
@@ -1,5 +1,115 @@
-window.onload = function() {
-  const canvas = document.getElementById('mainDrawingCanvas');
-  const context = canvas.getContext('2d');
+const validShapes = ['triangle', 'circle', 'rect'];
+const sin60 = Math.sqrt(3) / 2;
 
+window.onload = function () {
+    const canvas = document.getElementById('mainDrawingCanvas');
+    const context = canvas.getContext('2d');
+    const width = canvas.clientWidth;
+    const height = canvas.clientHeight;
+    const figures = ('shape:circle/center:120,310/radius:100/velocity:1,2/color:#64b1' +
+        '|shape:rect/center:256,128/width:78/height:154/velocity:2,3/color:#3381' +
+        '|shape:triangle/center:340,389/length:180/velocity:7,1/color:#19a1' +
+        '|shape:circle/center:500,500/radius:70/velocity:-4,-6/color:#b1b1' +
+        '|shape:rect/center:200,128/width:100/height:84/velocity:2,-4/color:#3441' +
+        '|shape:triangle/center:340,389/length:150/velocity:7,10/color:#49f1')
+        .split(' ').join('').split('|').map((value) => parseFigure(value))
+        .filter((value) => value && value.containsX(width) && value.containsY(height));
+    window.setInterval(function () {
+        context.clearRect(0, 0, width, height);
+        figures.forEach((figure) => {
+            figure.draw(context);
+            figure.center[0] += figure.velocity[0];
+            figure.center[1] += figure.velocity[1];
+            figure.checkIfContainsAndChangeDirection(width, height);
+        });
+    }, 50 / 3);
 };
+
+function parseFigure(string) {
+    const figure = {
+        color: '#000',
+        velocity: [2, 2],
+        changeDirection: function (index) {
+            this.velocity[index] = -this.velocity[index];
+        },
+        checkIfContainsAndChangeDirection: function (width, height) {
+            if (!this.containsX(width)) {
+                this.changeDirection(0);
+            }
+            if (!this.containsY(height)) {
+                this.changeDirection(1);
+            }
+        },
+    };
+    string.split('/').forEach((value) => {
+        const pair = value.split(':');
+        figure[pair[0]] = pair[0] === 'shape' || pair[0] === 'color' ? pair[1] : pair[1].includes(',') ?
+            pair[1].split(',').map((stringValue) => parseInt(stringValue)) : parseInt(pair[1]);
+    });
+    if (!validShapes.includes(figure.shape) || figure.center === undefined) {
+        return false;
```

Comment:
Caller will expect a figure or maybe null / undefined, but not a boolean, so it's better to return null or use an empty return statement `return;` (this will return undefined)

Reply:
done